src/oic/oic/provider.py

```diff
@@ -2027,3 +2027,12 @@ def remove_inactive_keys(self, more_then=3600):
                         kb.remove(key)
             if len(kb) == 0:
                 self.keyjar.issuer_keys[""].remove(kb)
+
+    def get_by_sub_and_(self, sub, key, val):
+        for sid in self.sdb.get_by_sub(sub):
```

Comment:
This should be annotated, also needs a docstring.

The `KeyError` and `return None` are not covered by tests.

Reply:
OK